linters-requirements.txt

```diff
@@ -0,0 +1,5 @@
+# linters have different requirements than test ones, some would
+# conflict, like ansible version required by ansible-lint.
+flake8
+yamllint>=1.23.0
+ansible-lint>=4.3.0a1,<5
```

Comment:
Is it ok to use alpha?

Reply:
currently having same requirements as upstream zuul-jobs, thus - should be fine. Actially 99% of this PR is copying over stuff from zuul-jobs, while the ansible-collections stuff comes from https://review.opendev.org/#/c/730360/ (seems to stuck for a while)